src/ui/public/visualize/loader/errors.ts

```diff
@@ -17,8 +17,22 @@
  * under the License.
  */
 
-import { ElasticsearchError } from './elasticsearch_error';
+import { RequestHandlerParams, Vis } from '../../vis';
 
-export function isTermSizeZeroError(error) {
-  return ElasticsearchError.hasRootCause(error, 'size must be positive, got 0');
+import { toastNotifications } from 'ui/notify';
+
+export function handleLoaderError(params: RequestHandlerParams, vis: Vis, error: any): void {
+  params.searchSource.cancelQueued();
+
+  vis.requestError = error;
+  vis.showRequestError =
+    error.type && ['NO_OP_SEARCH_STRATEGY', 'UNSUPPORTED_QUERY'].includes(error.type);
```

Comment:
I'm not sure if this is necessary either or how to reproduce errors of this type; would welcome any input here.